src/polymer/dom-calls-to-native.ts

```diff
@@ -0,0 +1,121 @@
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import babelTraverse from 'babel-traverse';
+import * as babel from 'babel-types';
+import {Document, Severity, Warning, Class, SourceRange, ParsedDocument, isPositionInsideRange} from 'polymer-analyzer';
+import {Function as AnalyzedFunction} from 'polymer-analyzer/lib/javascript/function';
+
+import {registry} from '../registry';
+import {Rule} from '../rule';
+import {stripWhitespace} from '../util';
+
+import {stripIndentation} from '../util';
+
+
+class DomCallsToNative extends Rule {
+  code = 'dom-calls-to-native';
+  description = stripIndentation(`
+      Warns when the Polymer.dom is used in places where native
+      methods can now be used instead.
+
+        Polymer.dom(event).path
+
+      Accepted syntax:
+
+        event.composedPath()
+  `);
+
+  private _replacements: Map<string, string> = new Map([
+    ['localTarget', 'target'],
+    ['rootTarget', 'composedPath()[0]'],
+    ['path', 'composedPath()']
+  ]);
+
+  async check(document: Document) {
+    const warnings: Warning[] = [];
+
+    const features = new Array<Class|AnalyzedFunction>(
+        ...document.getFeatures({kind: 'class'}),
+        ...document.getFeatures({kind: 'function'}));
+
+    for (const feature of features) {
+      babelTraverse(feature.astNode, {
+        noScope: true,
+        MemberExpression: (path) => {
+          if (!babel.isIdentifier(path.node.property)) {
+            return;
+          }
+
+          const name = path.node.property.name;
+          const doc = getParsedDocumentContaining(feature.sourceRange, document);
+
+          if (!doc) {
+            return;
+          }
+
+          const sourceRange = doc.sourceRangeForNode(path.node)!;
+
+          if (this._isPolymerDomCall(path.node) && this._replacements.has(name)) {
+            const replacement = this._replacements.get(name)!;
+
+            warnings.push(new Warning({
+              parsedDocument: document.parsedDocument,
+              code: 'deprecated-dom-call',
+              severity: Severity.WARNING,
+              sourceRange,
+              message: stripWhitespace(`
+                Polymer.dom no longer needs to be called for "${name}",
+                instead "event.${replacement}" may be used.
+              `)
+            }));
+          }
+        }
+      });
+    }
+
+    return warnings;
+  }
+
+  private _isPolymerDomCall(expr: babel.MemberExpression): boolean {
+    return babel.isCallExpression(expr.object) &&
+      babel.isMemberExpression(expr.object.callee) &&
+      babel.isIdentifier(expr.object.callee.object) &&
+      babel.isIdentifier(expr.object.callee.property) &&
+      expr.object.callee.object.name === 'Polymer' &&
+      expr.object.callee.property.name === 'dom';
+  }
+}
+
+registry.register(new DomCallsToNative());
+
+function getParsedDocumentContaining(
```

Comment:
this thing needs to go away... we have a method now in the analyzer (somewhere inside `Analysis`). but im not sure how we get at it...

Reply:
Yeah, we need to pass down the `analysis` to `checkDocument`. Mind putting a TODO and I'll tackle that in a followup PR?